libparsec/README.md

```diff
@@ -0,0 +1,161 @@
+the platform they are compiled for.
+
+For instance `libparsec_platform_mountpoint` uses FUSE on UNIX, WinFSP on Windows,
+and is not compatible with web.
+
+The Corollary is that crates without `platform` in their name are platform-independent,
+and hence can only be tested on a single platform.
+
+The platform crates:
+
+- `platform_async`: Asynchronous programming works differently between web and native
+  (the former has to integrate with the javascript runtime, the latter uses Tokio), so
+  this crate exposes platform-agnostic utilities (e.g. spawn a future, async lock etc.).
+- `platform_device_loader`: Load the device keys file from disk.
+- `platform_http_proxy`: Handle HTTP proxy.
+- `platform_ipc`: Allow communication between processes to ensure a device is not
+                  concurrently used.
+- `platform_mountpoint`: FUSE/WinFSP to expose the workspaces as a OS drive.
+- `platform_storage`: Database to store on disk manifest/block/certificates etc.
+
+### Test helpers crates
+
+Here, "testing" refers to two use-cases:
+
+1. Running tests on a crate (i.e. `cargo test -p my_crate`)
+2. Running the application with test feature (e.g. we want to run the GUI with an
+  dummy organization already configured for quick testing purpose)
+
+- `testbed`:
+- `tests_macros`: Defines the `#[parsec_test]` macro that is used to decorate each
+  test function. Among other things, this macro conveniently handles setting up
+  Tokio context (so that our test can be an asynchronous function) and the testbed
+  (configuring a test organization, and starting a testbed server if needed).
+- `tests_lite`: Re-expose utilities useful to write tests (e.g. `pretty_assertions`,
+  `rstest`, the `parsec_test` macro etc.). This is a lighter version of `tests_fixture`
+  that is needed to tests `types` crate (since `tests_fixture` internally uses `types`
+  crate).
+- `tests_fixtures`: This crate re-expose the content of `test_lite` and provides additional
+  helpers such as temporary folder fixture, injection hook, testbed types etc.
+
+> Notes:
+>
+> - From an external point of view, the `tests_fixture` re-expose everything needed.
+>   So when writing tests, `use libparsec_tests_fixtures::prelude::*` is the single-no-brainer-one-liner™.
+> - `tests_fixtures`'s name is mostly an historical artifact, as it contains far
+>   more than only fixtures !
+
+### Special crates
+
+- `crypto` Implement all the crypto algo (signing, encryption, hashing etc.). There is two
+  implementation for those: Rustcrypto (pure Rust) and Sodiumoxide (based on libsodium).
+- `zstd`: That's a funny one, you should have a look at [its README](./crates/zstd/README.md) ;-)
+
+## FAQ
+
+### Why most tests are in `tests/unit` ?
+
+#### TL;DR
+
+Stick to the following rule:
+
+- All tests are unit tests.
+- All test code must be in `tests/unit` (this works by using `#[path = "../../tests/unit/my_test.rs"]`
+  to overwrite the test module location).
```

Comment:
```suggestion
- All test code must be in `tests/unit` (this works by using `#[path = "../../tests/unit/my_test.rs"]`
  to overwrite the test module location in the tested crate, where you'd put your usual unit tests).
```

I'm not sure about the exact formulation, but what I'm trying to say is to explicit where to add the `#[path = "..."]` line

Reply:
I think we don't need extensive explanation here given this pattern is already used extensively in the code (so it's easy to just search for `#[path = ` in the codebase to understand how it is used).